addons/binding/org.openhab.binding.jeelink/src/main/java/org/openhab/binding/jeelink/handler/AbstractJeeLinkConnection.java

```diff
@@ -0,0 +1,137 @@
+    private final Logger logger = LoggerFactory.getLogger(AbstractJeeLinkConnection.class);
+
+    protected final ArrayList<JeeLinkReadingConverter<?>> inputHandlers = new ArrayList<>();
+    protected String sketchName = null;
+    protected String port;
+
+    private String[] initCommands;
+
+    public AbstractJeeLinkConnection(String port) {
+        this.port = port;
+    }
+
+    @Override
+    public String getPort() {
+        return port;
+    }
+
+    @Override
+    public void addReadingConverter(JeeLinkReadingConverter<?> listener) {
+        synchronized (inputHandlers) {
+            if (!inputHandlers.contains(listener)) {
+                logger.debug("Added listener {}.", listener);
+                inputHandlers.add(listener);
+            }
+        }
+    }
+
+    @Override
+    public void removeReadingConverters() {
+        synchronized (inputHandlers) {
+            inputHandlers.clear();
+        }
+    }
+
+    public void propagateLine(String line) {
+        logger.trace("Read line from port {}: {}", port, line);
+
+        if (sketchName == null) {
+            Matcher matcher = Pattern.compile("\\[(\\w+).*\\]").matcher(line);
+            boolean matches = matcher.matches();
+            if (matches) {
+                sketchName = matcher.group(1);
+                logger.debug("Sketch name found for port {}: {}", port, sketchName);
+
+                initializeDevice();
+            }
+        }
+
+        synchronized (inputHandlers) {
+            for (JeeLinkReadingConverter<?> l : inputHandlers) {
+                l.handleInput(line);
+            }
+        }
+    }
+
+    @Override
+    public void setInitCommands(String initCommands) {
+        if (initCommands != null && !initCommands.trim().isEmpty()) {
+            this.initCommands = initCommands.split(";");
+        }
+    }
+
+    private void initializeDevice() {
+        try {
+            if (initCommands != null) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Initializing device on port {} with commands {} ", port,
+                            Arrays.toString(initCommands));
+                }
+                OutputStream initStream;
+                initStream = getInitStream();
+                if (initStream != null) {
+                    // do not close the writer as this closes the underlying stream, and
+                    // in case of tcp connections, the underlying socket
+                    OutputStreamWriter w = new OutputStreamWriter(initStream);
+                    for (String cmd : initCommands) {
+                        w.write(cmd);
+                    }
+                } else {
+                    logger.warn("Connection on port {} did not provide an init stream for writing init commands", port);
+                }
+            }
+        } catch (IOException e) {
+            logger.error("Failed to send init commands to port {}: {}", port, e.getMessage());
```

Comment:
Don't log errors, but set the Thing status to OFFLINE with an according error message.